src/route/builder.rs

```diff
@@ -0,0 +1,377 @@
+// SPDX-License-Identifier: MIT
+
+use std::{
+    marker::PhantomData,
+    net::{IpAddr, Ipv4Addr, Ipv6Addr},
+};
+
+use netlink_packet_route::{
+    route::{
+        RouteAddress, RouteAttribute, RouteHeader, RouteMessage, RouteProtocol,
+        RouteScope, RouteType,
+    },
+    AddressFamily,
+};
+
+pub struct RouteMessageBuilder<T = IpAddr> {
+    message: RouteMessage,
+    _phantom: PhantomData<T>,
+}
+
+impl<T> RouteMessageBuilder<T> {
+    fn new_no_address_family() -> Self {
+        let mut message = RouteMessage::default();
+        message.header.table = RouteHeader::RT_TABLE_MAIN;
+        message.header.protocol = RouteProtocol::Static;
+        message.header.scope = RouteScope::Universe;
+        message.header.kind = RouteType::Unicast;
+        Self {
+            message,
+            _phantom: Default::default(),
+        }
+    }
+
+    /// Sets the input interface index.
+    pub fn input_interface(mut self, index: u32) -> Self {
+        self.message.attributes.push(RouteAttribute::Iif(index));
+        self
+    }
+
+    /// Sets the output interface index.
+    pub fn output_interface(mut self, index: u32) -> Self {
+        self.message.attributes.push(RouteAttribute::Oif(index));
+        self
+    }
+
+    /// Sets the route priority (metric)
+    pub fn priority(mut self, priority: u32) -> Self {
+        self.message
+            .attributes
+            .push(RouteAttribute::Priority(priority));
+        self
+    }
+
+    /// Sets the route table.
+    ///
+    /// Default is main route table.
+    #[deprecated(note = "Please use `table_id` instead")]
```

Comment:
Since we are breaking the API by replace it with new. There is no need to keep old deprecated function in new struct.

Reply:
Removed, thanks!